### PR TITLE
Allow {url} and {url:pretty} as parts of arguments

### PR DESCRIPTION
**Pull request by blyxxyz**
This makes commands like `:open web.archive.org/web/{url}` possible.

It also adds a no_replace_variables command register argument that stops the replacement from happening, which is important for commands like `:bind` and `:spawn` that take a command as an argument.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Maybe we should also add something to `src2asciidoc.py` so the fact that the command doesn't replace variables is noted in `commands.asciidoc` like `no_cmd_split` and `maxsplit` are?

File: qutebrowser/commands/command.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ class Command:
         no_cmd_split: If true, ';;' to split sub-commands is ignored.
         backend: Which backend the command works with (or None if it works with
                  both)
+        no_replace_variables: Whether or not to replace variables like {url}
```
Comment (The-Compiler) [inline qutebrowser/commands/command.py]: I think "Don't replace variables ..." would be cleaner.

File: qutebrowser/mainwindow/statusbar/command.py [hunk 2]
```diff
@@ -109,7 +109,6 @@ def set_cmd_text_command(self, text, space=False, append=False):
             append: If given, the text is appended to the current text.
         """
         args = split.simple_split(text)
-        args = runners.replace_variables(self._win_id, args)
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/command.py]: I'm a bit lost here - why was this removed?

Comment (blyxxyz) [inline qutebrowser/mainwindow/statusbar/command.py]: That line was there because the variable replacement would otherwise only happen if `{url}` or `{url:pretty}` were the full command text, but that's no longer a problem. I should remove the two lines surrounding it as well, I missed that.

File: scripts/dev/src2asciidoc.py [hunk 3]
```diff
@@ -248,6 +248,8 @@ def _get_command_doc_notes(cmd):
         if cmd.no_cmd_split:
             yield ("* With this command, +;;+ is interpreted literally "
                    "instead of splitting off a second command.")
+        if cmd.no_replace_variables:
+            yield "* This command does not replace variables like +\{url\}+."
```
Comment (The-Compiler) [inline scripts/dev/src2asciidoc.py]: Please use a raw string (`r"..."`) here or double the backslashes (I'm guessing you're using them because `{` and `}` have some special meaning in asciidoc?)

File: qutebrowser/browser/commands.py [hunk 4]
```diff
@@ -992,7 +992,7 @@ def tab_move(self, index: typing.Union[str, int]=None, count=None):
             self._tabbed_browser.setUpdatesEnabled(True)
 
     @cmdutils.register(instance='command-dispatcher', scope='window',
-                       maxsplit=0)
+                       maxsplit=0, no_replace_variables=True)
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: Hmm, while this makes sense (as the command needs to replace them *after* splitting the commandline) this adds the respective line to the documentation, which is really confusing... Not sure what's the best way around this though?

Comment (blyxxyz) [inline qutebrowser/browser/commands.py]: I removed maxsplit and no_replace_variables to let the regular parser handle things.

**Comment by The-Compiler** [comment]
I don't remember the details off-hand, but as you can see with the failing tests, there was a reason the splitting was done manually there :wink:

**Comment by blyxxyz** [comment]
There are some failures because I didn't check imports and tests to adjust, but the hardest problem to solve is that it treats shell command options as qutebrowser command options. I expect there's no simple solution for that.

Maybe the best thing to do is just to add a line to `spawn`'s documentation saying that the variable substitution is handled by the command itself.

**Comment by The-Compiler** [comment]
I was wondering about that too, but then the documentation contradicts itself. Maybe we should just explicitly blacklist `spawn` in `src2asciidoc` - seems like the least ugly solution to me.

**Comment by The-Compiler** [comment]
Thanks!
